ui/models/runnableJob.js

```diff
@@ -1,15 +1,46 @@
-import { getProjectUrl } from '../helpers/url';
+import { getProjectUrl, getRunnableJobsURL } from '../helpers/url';
+import { escapeId } from '../helpers/aggregateId';
 
 import JobModel from './job';
 
-const uri = getProjectUrl('/runnable_jobs/');
-
 export default class RunnableJobModel {
   constructor(data) {
     Object.assign(this, data);
   }
 
-  static getList(repoName, params) {
-    return JobModel.getList(repoName, params, { uri });
+  static async getList(repoName, params) {
+    let uri = getRunnableJobsURL(params.decision_task_id);
+    let rawJobs = await fetch(uri).then(response => response.json());
+    const jobList = [];
+
+    // TODO: Remove this fallback once the gz artifacts expire
+    if (rawJobs.code === 'ResourceNotFound') {
+      uri = getProjectUrl('/runnable_jobs/');
+      rawJobs = await JobModel.getList(repoName, params, { uri });
+      rawJobs.forEach(job => {
+        job.push_id = params.push_id;
+        job.id = escapeId(params.push_id + job.ref_data_name);
+      });
```

Comment:
I'm hoping it's okay to move appending the push_id and id into each job into here from Push.jsx? 

It just avoids having to iterate through all 11,000+ jobs multiple times.